Community/Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC.js

```diff
@@ -0,0 +1,400 @@
+          response.preset = `, -map 0 -c copy ${extraArguments}`;
+          response.processFile = true;
+          return response;
+        }
+        // New logic for reprocessing HEVC. Mainly done for my own use. Since we're reprocessing we're checking
+        // bitrate again and since this can be inaccurate (It calculates overall bitrate not video specific)
+        // we have to inflate the current bitrate so we don't keep looping this logic.
+      } else if (inputs.reconvert_hevc === 'true' && (file.ffProbeData.streams[i].codec_name === 'hevc'
+        || file.ffProbeData.streams[i].codec_name === 'vp9' || file.ffProbeData.streams[i].codec_name === 'av1')) {
+        // If we're using the hevc max bitrate then update the cutoff to use it
+        if (inputs.hevc_max_bitrate !== '') {
+          if (currentBitrate > inputs.hevc_max_bitrate) {
+            inflatedCutoff = Math.round(inputs.bitrate_cutoff);
+            response.infoLog += `☒ Reconvert_hevc is ${inputs.reconvert_hevc} & the file is already hevc, vp9 or av1. 
+            Using hevc specific cutoff of ${inputs.hevc_max_bitrate}. 
+            \n☒ The file is still above this new cutoff! Reconverting. \n`;
+          } else {
+            response.processFile = false;
+            inflatedCutoff = Math.round(inputs.bitrate_cutoff);
+            response.infoLog += `☑ Reconvert_hevc is ${inputs.reconvert_hevc} & the file is already hevc, vp9 or av1. 
+            Using hevc specific cutoff of ${inputs.hevc_max_bitrate}. 
+            \n☑ The file is NOT above this new cutoff. Exiting plugin. \n`;
+            return response;
+          }
+          // If we're not using the hevc max bitrate then we need a safety net
+          // to try and ensure we don't keep looping this plugin.
+        } else if (currentBitrate > (inputs.bitrate_cutoff * 2)) {
+          inflatedCutoff = Math.round((inputs.bitrate_cutoff * 2));
+          response.infoLog += `☒ Reconvert_hevc is ${inputs.reconvert_hevc} & the file is already hevc, vp9 or av1. 
+          hevc specific cutoff not set so bitrate_cutoff is multiplied by 2 for safety! 
+          Cutoff now temporarily ${inflatedCutoff}. \n The file is still above this new cutoff! Reconverting. \n\n`;
+        } else {
+          response.processFile = false;
+          inflatedCutoff = Math.round((inputs.bitrate_cutoff * 2));
+          response.infoLog += `☑ Reconvert_hevc is ${inputs.reconvert_hevc} & the file is already hevc, vp9 or av1 
+          so bitrate_cutoff is multiplied by 2! Cutoff now temporarily ${inflatedCutoff}. 
+          \n The file is NOT above this new cutoff. Exiting plugin. \n\n`;
+          return response;
+        }
+      }
+    }
+
+    // Are we encoding 10 bit files?
+    if (inputs.enable_10bit === 'true') {
+      // Check if video stream is 10bit. Currently set profile main10 and pixel format.
+      // Still need to test if this really works correctly
```

Comment:
If it needs testing, why is it in a production plugin? :)

Reply:
Was mainly referring to HDR being untested rather than 10 bit. Now there is a HDR check which is tested and should catch HDR files.